eth-rpc/server.nim

```diff
@@ -135,16 +136,73 @@ proc processClient(server: RpcServer, client: AsyncSocket) {.async.} =
         let err = future.readError[].ValueError
         await client.sendError(INVALID_PARAMS, err.msg, %"")
       else:
-        await client.sendError(SERVER_ERROR, "Error: Unknown error occurred", %"")
+        await client.sendError(SERVER_ERROR,
+                               "Error: Unknown error occurred", %"")
 
-proc serve*(server: RpcServer) {.async.} =
+proc newRpcServer*(address = "localhost", port: Port = Port(8545)): RpcServer =
```

Comment:
Using a `seq[IpAddress]` type instead of string is better here. The only valid value for a string that is not an ip address is "localhost", which can easily be handled elsewhere. While `strings` interface limits the user in available configurations. E.g. the server can't be configured to listen on _all_ external interfaces, can it?

This interface could be left as a thin shortcut to a more elaborate interface, but I'm not really sure if that's needed.

Reply:
`address = "0.0.0.0"` all external interfaces via IPv4.
But please note this is was before this PR, so i just fixed behavior of `address: string`.

Reply:
Yes, you can't listen on all external interfaces on both ipv4 and ipv6. Since you're changing the interface anyway, why not change it to smth even better.

Reply:
@yglukhov, please, I have not changed any interface of RPC.

Reply:
Ah, right. Ok.

Reply:
I guess @yglukhov is making a valid point that you might want to listen on multiple distinct interfaces (not just all of them). We can try to support such a `seq` parameter throughout the whole networking stack.

Reply:
@zah, true, although as @cheatfate says this could be a subject for another pr.

Reply:
These changes can be part of the refactoring to support network agnostic transports.